core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ActiveDirectoryUserStoreManager.java

```diff
@@ -566,7 +565,7 @@ public void doSetUserClaimValues(String userName, Map<String, String> claims, St
                     updatedAttributes);
 
             if (cnModified && cnValue != null) {
-                subDirContext.rename(returnedUserEntry, "CN=" + escapeUsernameSpecialCharacters(cnValue, true));
+                subDirContext.rename(returnedUserEntry, "CN=" + escapeSpecialCharactersForDN(cnValue));
             }
 
         } catch (Exception e) {
```

Comment:
changed the code to catch the specific exceptions in https://github.com/wso2/carbon4-kernel/compare/master...nuwandiw:refac#diff-99f5dfe4a22bacd7db1ad0367628376fR560